### PR TITLE
ZEN-28165: Instrument zenmodeler for self-monitoring

### PR DESCRIPTION
**Pull request by youngklee**
modeledDevices and the number of failures are kept track of.

### CODE REVIEW

File: Products/DataCollector/zenmodeler.py [hunk 1]
```diff
@@ -61,6 +61,8 @@
 import traceback
 from random import randint
 from itertools import chain
+from metrology import Metrology
+from metrology.instruments import Gauge
```
Comment (jplouis) [inline Products/DataCollector/zenmodeler.py]: This doesn't need to be imported

File: Products/DataCollector/zenmodeler.py [hunk 2]
```diff
@@ -710,10 +715,12 @@ def processClientFinished(result):
             @type result: object
             """
             self.counters['modeledDevicesCount'] += 1
+            self._modeledDevicesMetric.mark(self.counters['modeledDevicesCount'])
```
Comment (jplouis) [inline Products/DataCollector/zenmodeler.py]: You mark a meter with the delta, not an entire new value

File: Products/DataCollector/zenmodeler.py (hunk 1 above)
Comment (youngklee) [inline Products/DataCollector/zenmodeler.py]: Gauge doesn't need to be, but Metrology does.

File: Products/DataCollector/zenmodeler.py (hunk 2 above)
Comment (youngklee) [inline Products/DataCollector/zenmodeler.py]: Revised as suggested.

File: Products/DataCollector/zenmodeler.py (hunk 1 above)
Comment (youngklee) [inline Products/DataCollector/zenmodeler.py]: Removed import Gauge

**Pull request merged**